src/engine/QueryPlanningCostFactors.cpp

```diff
@@ -24,14 +26,16 @@ QueryPlanningCostFactors::QueryPlanningCostFactors() : _factors() {
 
 // _____________________________________________________________________________
 void QueryPlanningCostFactors::readFromFile(const string& fileName) {
-  std::ifstream in(fileName.c_str());
+  std::ifstream in(fileName);
   string line;
   while (std::getline(in, line)) {
-    auto v = ad_utility::split(line, '\t');
+    std::vector<std::string_view> v = absl::StrSplit(line, '\t');
     AD_CHECK_EQ(v.size(), 2);
+    float factor;
+    std::from_chars(v[1].data(), v[1].data() + v[1].size(), factor);
```

Comment:
I don't think, that this is often used,
But still, from_chars returns a status which should be checked.